tests-wtr/dom-actions/dom-actions.element-selector.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from '@esm-bundle/chai';
+import { utils } from "./utils";
+
+// must be run from dom-actions.element-selector.html
+describe('elementSelector', () => {
+  describe('querySingleReplySelector', () => {
+    const specialCasesThatShouldReturnEmptyArray = ['aria/', 'text/', 'pierce/']
+
+    specialCasesThatShouldReturnEmptyArray.forEach((selector) => {
+      it(`should return empty array if selector starts with "${selector}"`, () => {
+        // Given
+        const domActions = utils.instantiateDomActions();
+
+        // When
+        const elements = domActions.querySingleReplySelector(selector + 'something');
+
+        // Then
+        expect(elements).to.be.empty
+      })
+    })
+    it('should return elements by xpath if selector starts with "xpath/"', () => {
+      // Given
+      const domActions = utils.instantiateDomActions();
+
+      // When
+      const elements = domActions.querySingleReplySelector('xpath///button[contains(., \'Accept all\')]');
+
+      // Then
+      expect(elements).to.deep.eq([
+        document.getElementById('button1'),
+        document.getElementById('button3'),
+      ])
+    })
+    it('should return elements by querySelectorAll from shadow root if available', () => {
+      // Given
+      const domActions = utils.instantiateDomActions();
+
+      const shadowDiv = document.createElement('div')
+      document.body.appendChild(shadowDiv)
+      const shadow = shadowDiv.attachShadow({mode: 'open'})
+      const shadowButton1 = document.createElement('button')
+      shadow.appendChild(shadowButton1)
+      const shadowButton2 = document.createElement('button')
+      shadow.appendChild(shadowButton2)
+
+      // When
+      const elements = domActions.querySingleReplySelector('button', shadowDiv)
+
+      // Then
+      expect(elements).to.deep.eq([
+        shadowButton1,
+        shadowButton2,
+      ])
+    })
+    it('should return elements by querySelectorAll', () => {
+      // Given
+      const domActions = utils.instantiateDomActions();
+
+      // When
+      const elements = domActions.querySingleReplySelector('button');
+
+      // Then
+      expect(elements).to.deep.eq([
+        document.getElementById('button1'),
+        document.getElementById('button2'),
+        document.getElementById('button3'),
+      ])
+    })
+  })
+  describe('querySelectorChain', () => {
```

Comment:
could you add a test for Shadow DOM? something for `['.my-shadow-root-selector', '.selector-inside-shadow-root']`?

Reply:
good call! I'll adjust the `should return elements by querySelectorAll from shadow root if available` which now cannot be used directly anymore as the elementSelector() only accepts one argument.